Src/DependencyCollector/Shared/Implementation/TelemetryDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,293 @@
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+    using Microsoft.ApplicationInsights.Channel;
+
+    internal class TelemetryDiagnosticSourceListener : IObserver<DiagnosticListener>, IDisposable
+    {
+        private const string ActivityNameSuffix = ".Monitoring";
```

Comment:
I still prefer to get rid of it :)

Reply:
the only problem is that in that case we have to subscribe to all events - not only activities. Why? because typically before activity starts (so ".Start" event is raised) the code will check if activity is enabled - and that activity name will not have anything we can filter by so we need to subscribe to all events to get a chance to capture ".Start" events

...unless I'm missing sth

Reply:
You can just return false to all events that end with Start and thus enable all possible events.

Now, the 'monitoring' suffix you have, tells to the listener that it's going to be about Activity.
If we are going to filter sources that are compatible with Activity based on something, it should rather be a DiagnosticSource name than Activity name. It would be more efficient (as long names need to be compared only when someone subscribes to a source rather than per each operation).

I assume you want to support scenarios when DiagnosticSource sends both Activity Start/Stop and other events.
Our guidance should be to discourage this scenario at all. As we discussed offline, DiagnosticSource is not a logger.
In the worst case, we let in all other events and filter them out only in the Write-callback

I think once we have a guidance for lib owners, we should review it by few more people (including some lib owners)  and then we can return back to the listener name-based filtering.

Reply:
ok, I'll remove this 
good point about filtering DS vs Activities - with DS white/blacklist that should be flexible enough